music_assistant/common/models/media_items.py

```diff
@@ -465,28 +470,22 @@ class AlbumTrack(Track):
 
     @classmethod
     def from_track(
-        cls: type,
+        cls,
         track: Track,
         album: Album | None = None,
         disc_number: int | None = None,
         track_number: int | None = None,
-    ) -> Self:
+    ) -> AlbumTrack:
```

Comment:
Again, mypy seems to struggle a bit with classmethods and self. Self is apparently meant for when there might be a subclass, so i'm not too worried that I had to set the type here.

Reply:
ok strange, as far as I remember the Self type was explcitly added for this case. But fine to just use the class as type but I wonder why they introduced the Self type then

Reply:
The error you get is 

```
error: Incompatible return value type (got "AlbumTrack", expected "Self")  [return-value]
```

Possibly https://github.com/python/mypy/issues/16558?

Which says, if you subclassed `AlbumTrack` and `SomeSubclass.from_track` returned an `AlbumTrack`, rather than `SomeSubclass`, then there would be a type violation.

I ran into this after a similar problem where the mypy issue said to just not use Self for enums (iirc), so assumed this one was similar. But with fresh eye, it also looks like it works to do `cls.from_dict()`. 

One of the issues said `Self` was intended for cases where you were subclassing, as a shortcut for making a method Generic[] over cls or self etc.

Reply:
OK all good then, I actually like the explicit typing more than Self ;-)